### PR TITLE
Allow the creation of haproxy template directly on vSphere

### PR DESCRIPTION
**Pull request by marema31**
**What this PR does / why we need it**: Allow the creation of haproxy template directly on the vSphere using the new vsphere-iso provisioner of packer (using API and no more ssh hack)

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/tools/haproxy/README.md [hunk 1]
```diff
@@ -208,3 +209,40 @@ Or, to produce a list of URLs for the same image files (and their checksums), th
...
+
+## Template
+
+In production the haproxy loadbalancer is deployed as an template.
+
+### Template Requirements
+
+Building the template requires:
+
+* VMware vSphere >= 6.5
```
Comment (yastij) [inline hack/tools/haproxy/README.md:221]: we currently do not test against anything less than vsphere 6.7u3. I'd suggest bumping this to 6.7u3

File: hack/tools/haproxy/Makefile [hunk 2]
```diff
@@ -68,6 +68,14 @@ build: ## Builds the HAProxy load balancer OVA
 verify: ## Verifies the packer config
 	packer validate $(PACKER_FLAGS) packer.json
 
+.PHONY: build-template
+build-template: ## Builds the HAProxy load balancer OVA
```
Comment (yastij) [inline hack/tools/haproxy/Makefile:72]: vsphere-build-template

File: hack/tools/haproxy/Makefile [hunk 3]
```diff
@@ -68,6 +68,14 @@ build: ## Builds the HAProxy load balancer OVA
...
+	packer build $(PACKER_FLAGS) packer-template.json
+
+.PHONY: verify-template
+verify-template: ## Verifies the packer config
```
Comment (yastij) [inline hack/tools/haproxy/Makefile:76]: vsphere-verify-template

File: hack/tools/haproxy/README.md (hunk 1 above)
Comment (marema31) [reply hack/tools/haproxy/README.md:221]: Ok, for the moment I'm only able to test with vsphere 6.5,  I should be soon be able to test with vSphere 6.7u3 and 6.5, do you prefer me to bump it ?

**Comment by marema31** [comment]
I've done the asked modification, but I was wondering if you prefer me  to merge OVA and vsphere-template packer files and do export @codenrhoden  ?

**Comment by randomvariable** [comment]
@marema31 , let's focus on https://github.com/kubernetes-sigs/image-builder/pull/186 first.

As per #884, we may move all this to the image builder project anyway.

**Comment by codenrhoden** [comment]
yeah, I wouldn't pursue this here at all. I'm going to move this all over to image-builder hopefully in the next week. :) But a huge thanks to @marema31 for taking on the work in `image-builder` for adding the vCenter based builder!

**Comment by randomvariable** [comment]
Hi @marema31 .

Thanks for all the work on getting vSphere support into image builder. We're going to rally on getting the haproxy image into that project, so am going to close it.

Thannks!

/close

**Comment by marema31** [comment]
No problem, if you need help for this to happen or to test it ping me, my daywork give me access to playground vSphere infrastucture that I can use for this kind of activity

**Pull request closed without merging**